FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +20,50 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let _ = self else { return }
+			switch result {
+			case let .success((data, response)):
+				guard response.statusCode == RemoteFeedLoader.successStatusCode else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+				guard let feed = try? JSONDecoder().decode(Feed.self, from: data) else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+				completion(.success(feed.feedImages))
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+
+	private struct Feed: Decodable {
+		let items: [FeedImageAPI]
+
+		var feedImages: [FeedImage] {
+			items.compactMap { $0.feedImage }
+		}
+	}
+
+	private struct FeedImageAPI: Decodable {
+		let id: UUID
+		let description: String?
+		let location: String?
+		let urlString: String
+
+		enum CodingKeys: String, CodingKey {
+			case id = "image_id"
+			case description = "image_desc"
+			case location = "image_loc"
+			case urlString = "image_url"
+		}
+
+		var url: URL? { URL(string: urlString) }
+
+		var feedImage: FeedImage? {
+			url != nil ? FeedImage(id: id, description: description, location: location, url: url!) : nil
+		}
```

Comment:
Instead of using a force-unwrap, the optionality of `url` could be checked with a `guard let url = url` statement.

But following from my previous comments, this helper can be non-optional, since `JSONDecoder` will guarantee that `url` is a valid `URL`:
```suggestion
		var feedImage: FeedImage {
			FeedImage(id: id, description: description, location: location, url: url)
		}
```